dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/icon/IconControllerTest.java

```diff
@@ -107,30 +104,44 @@ void shouldUpdateIconWhenKeyExists() throws IOException {
     assertEquals(String.format("Icon %s updated", iconKey), response.getString("message").string());
   }
 
+  @Test
+  void shouldGetOnlyCustomIcons() throws IOException {
+    String fileResourceId = createFileResource();
+    createIcon(fileResourceId, keywords);
+
+    JsonArray iconArray = GET("/icons?type=CUSTOM").content(HttpStatus.OK);
+
+    verifyAll(iconArray, keywords, fileResourceId);
+  }
+
+  @Test
+  void shouldGetIconsWithSpecifiedKey() throws IOException {
+    String fileResourceId = createFileResource();
+    createIcon(fileResourceId, keyword);
+
+    JsonArray iconArray = GET("/icons?keywords=m1").content(HttpStatus.OK);
+
+    verifyAll(iconArray, keyword, fileResourceId);
+  }
+
   @Test
   void shouldDeleteIconWhenKeyExists() throws IOException {
-    createIcon(createFileResource());
+    createIcon(createFileResource(), keywords);
 
     JsonObject response = DELETE(String.format("/icons/%s", iconKey)).content();
 
     assertEquals(String.format("Icon %s deleted", iconKey), response.getString("message").string());
   }
 
-  private String createIcon(String fileResourceId) {
+  private String createIcon(String fileResourceId, String keywords) {
```

Comment:
Do I understand correctly that a user can only create a custom icon? If so I would name this `createCustomIcon` to make this obvious. This makes it easier to understand some of the test cases 😅

Reply:
fixed